sql/core/src/test/scala/org/apache/spark/sql/execution/HiveResultSuite.scala

```diff
@@ -23,7 +23,7 @@ class HiveResultSuite extends SharedSparkSession {
   import testImplicits._
 
   test("date formatting in hive result") {
-    val dates = Seq("2018-12-28", "1582-10-13", "1582-10-14", "1582-10-15")
+    val dates = Seq("2018-12-28", "1581-10-13", "1581-10-14", "1581-10-15")
```

Comment:
I had to change the dates because I faced to a strange behavior of JDK 8 when I try to convert a local timestamp in Gregorian calendar to a local timestamp in Julian calendar. Switching from Julian to Gregorian must happen on 1582-10-15 but getCalendarSystem in Date.java considers entire year 1582 as Gregorian one:
https://github.com/frohoff/jdk8u-dev-jdk/blob/master/src/share/classes/java/util/Date.java#L1277-L1282
The test works for:
```
1581-10-13 01:02:03
1583-10-13 01:02:03
```
but not for:
```
1582-10-13 01:02:03
```
Another method of Date works more precisely
https://github.com/frohoff/jdk8u-dev-jdk/blob/master/src/share/classes/java/util/Date.java#L1284-L1294
but it is not used in the constructor:
https://github.com/frohoff/jdk8u-dev-jdk/blob/master/src/share/classes/java/util/Date.java#L253

Reply:
Converting dates between 1582-10-04 and 1582-10-15 seems ambiguous, and depends on the implementation. I changed to the dates to have more deterministic behavior.